src/bootstrap/lib.rs

```diff
@@ -846,18 +846,19 @@ impl Build {
         INTERNER.intern_path(self.out.join(&*target.triple).join("md-doc"))
     }
 
-    /// Returns `true` if no custom `llvm-config` is set for the specified target.
+    /// Returns `true` if this is our custom, patched, version of LLVM.
     ///
-    /// If no custom `llvm-config` was specified then Rust's llvm will be used.
+    /// This does not necessarily imply that we're managing the `llvm-project` submodule.
     fn is_rust_llvm(&self, target: TargetSelection) -> bool {
         match self.config.target_config.get(&target) {
+            // We're using a pre-built version of LLVM, but the user has promised that pre-built version has our patches.
```

Comment:
We may not be using a pre-built version of LLVM. Specifically the use case I want to support is cd into `src/llvm-project`, check out a vanilla commit, and set `submodules = false`, `llvm-has-rust-patches = false` in config.

Reply:
That seems totally bizarre, though. Why would you want that?

Reply:
To build against a more recent version of LLVM.

If you separately build and supply your own llvm-config, then you hit the other assumptions in here about not needing to distribute the LLVM libraries because it's probably system installed. So basically this was unsupported before.

Plus it's desirable to have bootstrap manage the configuration of LLVM if all you want to vary is the version.